lib/bank_account/checking_account.rb

```diff
@@ -0,0 +1,46 @@
+module BankAccount
+  class CheckingAccount < Account
+
+    def initialize(id, initial_balance)
+      # everything but the @check_counter instance variable is inherited from account parent class
+      @check_counter = 0
+      super
+    end
+
+    def withdraw(amount)
+
+      if amount > @balance
+        puts "Invalid. That amount is greater than your account balance."
+      else
+        super
+        new_balance = @balance - 1 # $1 transaction fee tacked on
+        @balance = new_balance
+      end
+      return @balance
```

Comment:
Your use of `super` is spot on here, but I think you could do without the `new_balance` local variable. This is also a good candidate for a guard clause like in my comment above.